### PR TITLE
Add vulnerability detection related content

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/21743 |

## Description

Adds the new vulnerability detection configuration to the AIT tests and restores the vulnerability detection fields that were deleted as part of https://github.com/wazuh/wazuh/pull/21542.

## Tests

https://github.com/wazuh/wazuh/issues/21743#issuecomment-1927770894

> The AIT tier 0/1 checks are failing because of https://github.com/wazuh/wazuh/issues/21803. I suspect it has to be with the workflow using cached steps.

### CODE REVIEW

File: api/api/spec/spec.yaml [hunk 1]
```diff
@@ -4133,6 +4133,8 @@ components:
             $ref: '#/components/schemas/LogSummary'
           wazuh-modulesd:syscollector:
             $ref: '#/components/schemas/LogSummary'
+          wazuh-modulesd:vulnerability-detection:
```
Comment (fdalmaup) [inline api/api/spec/spec.yaml:4136]:
```suggestion
          wazuh-modulesd:vulnerability-scanner:
```
Checked the following issue for references regarding the VD logs: https://github.com/wazuh/wazuh/issues/21789#issuecomment-1932540280 and also launched a manager with the PR's branch:
```
2024/02/08 10:37:33 wazuh-modulesd:vulnerability-scanner: INFO: Starting vulnerability_scanner module.
```

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — DISMISSED** [review]
LGTM

File: api/api/spec/spec.yaml [hunk 2]
```diff
@@ -4133,6 +4133,8 @@ components:
             $ref: '#/components/schemas/LogSummary'
           wazuh-modulesd:syscollector:
             $ref: '#/components/schemas/LogSummary'
+          wazuh-modulesd:vulnerability-scanner:
+            $ref: '#/components/schemas/LogSummary'
```
Comment (Selutario) [inline api/api/spec/spec.yaml:4137]: Indeed, there are some new additional categories inside `wazuh-modulesd` (for example `wazuh-modulesd:content_manager`) and outside (`indexer-connector`) which we could add too. This is a snippet of my 4.8.0 local env:
```
2024/02/09 09:53:37 wazuh-modulesd:content_manager: INFO: Starting content_manager module.
2024/02/09 09:53:37 wazuh-modulesd:database: INFO: Module started.
2024/02/09 09:53:37 wazuh-modulesd:download: INFO: Module started.
2024/02/09 09:53:37 wazuh-modulesd:control: INFO: Starting control thread.
2024/02/09 09:53:37 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/02/09 09:53:37 wazuh-modulesd:syscollector: INFO: Module started.
2024/02/09 09:53:37 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/02/09 09:53:37 indexer-connector: WARNING: No username and password found in the keystore, using default values.
2024/02/09 09:53:37 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2024/02/09 09:53:37 indexer-connector: WARNING: Error initializing IndexerConnector: Couldn't connect to server, we will try again after 2 seconds.
2024/02/09 09:53:37 wazuh-modulesd:vulnerability-scanner: ERROR: VulnerabilityScannerFacade::start: Couldn't find column family: 'vendor_map'
2024/02/09 09:53:37 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2024/02/09 09:53:37 wazuh-syscheckd: INFO: FIM sync module started.
2024/02/09 09:53:39 indexer-connector: WARNING: Error initializing IndexerConnector: Couldn't connect to server, we will try again after 4 seconds.
2024/02/09 09:53:42 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/02/09 09:53:42 sca: INFO: Security Configuration Assessment scan finished. Duration: 5 seconds.
2024/02/09 09:53:43 indexer-connector: WARNING: Error initializing IndexerConnector: Couldn't connect to server, we will try again after 8 seconds.
2024/02/09 09:53:51 indexer-connector: WARNING: Error initializing IndexerConnector: Couldn't connect to server, we will try again after 16 seconds.
2024/02/09 09:53:54 rootcheck: INFO: Ending rootcheck scan.
2024/02/09 09:54:07 indexer-connector: WARNING: Error initializing IndexerConnector: Couldn't connect to server, we will try again after 32 seconds.
```

**Review by Selutario — CHANGES_REQUESTED** [review]
GJ, just a small request. 

Also, there are some failing [checks](https://github.com/wazuh/wazuh/actions/runs/7831004209/job/21366350144?pr=21747) related to pyJWT being imported in 4.8.0 branch but not existing (yet) in deps 25. We can ignore it since it was already reported and addressed in:
- https://github.com/wazuh/wazuh/issues/21803

**Review by Selutario — APPROVED** [review]
LGTM. The failed checks are expected as mentioned [here](https://github.com/wazuh/wazuh/pull/21747#pullrequestreview-1872098693). A successful build was already shared in https://github.com/wazuh/wazuh/issues/21743#issuecomment-1927770894

**Pull request merged**